cirq/sim/state.py

```diff
@@ -121,3 +123,139 @@ def validate_normalized_state(state: np.ndarray, num_qubits: int,
     norm = np.sum(np.abs(state) ** 2)
     if not np.isclose(norm, 1):
         raise ValueError('State is not normalized instead had norm %s' % norm)
+
+
+def sample_state(
+    state: np.ndarray,
+    indices: List[int],
+    repetitions: int=1) -> List[List[bool]]:
+    """Samples repeatedly from measurements in the computational basis.
+
+    Note that this does not modify the passed in state.
+
+    Args:
+        state: The state to be measured. This state is assumed to be normalized.
+            The state must be of size 2 ** integer.  The state can be of shape
+            (2 ** integer) or (2, 2, ..., 2).
+        indices: Which qubits are measured. The state is assumed to be supplied
+            in big endian order. That is the xth index of v, when expressed as
+            a bitstring, has the largest values that the 0th index.
+        repetitions: The number of times to sample the state.
+
+    Returns:
+        Measurement results with True corresponding to the |1> state.
+        The outer list is for repetitions, and the inner corresponds to
+        measurements ordered by the input indices.
+
+    Raises:
+        ValueError if repetitions is less than one or size of state is not a
+            power of 2.
+        IndexError if the indices are out of range for the number of qubits
+            corresponding to the state.
+    """
+    if repetitions < 1:
+        raise ValueError(
+            'Number of repetitions cannot be negative. Was {}'.format(
+                repetitions))
+    num_qubits = _validate_num_qubits(state)
+    _validate_indices(num_qubits, indices)
+
+    if len(indices) == 0:
+        return [[]]
+
+    # Calculate the measurement probabilities.
+    probs = _probs(state, indices, num_qubits)
+
+    # We now have the probability vector, correctly ordered, so sample over
+    # it. Note that we us ints here, since numpy's choice does not allow for
+    # choosing from a list of tuples or list of lists.
+    result = np.random.choice(2 ** len(indices), size=repetitions, p=probs)
+    # Convert to bools and rearrange to match repetition being the outer list.
+    return np.transpose([(1 & (result >> i)).astype(np.bool) for i in
+                         range(len(indices))]).tolist()
+
+
+def measure_state(
```

Comment:
`measure_state_vector`?

Note that users will likely see this in the autocomplete list when trying to find measurement operations for the first time. And they might try `cirq.measure(vector)` when trying to trigger this functionality. **We should either make that work, or throw a relevant error message when they pass in an np.ndarray instead of qubits to `cirq.measure`.**

If this doesn't actually affect the value of the input, it would better be to call it something like `measured_state_vector` or `state_vector_after_measurement`.